src/transactions/feed/queryHelper.ts

```diff
@@ -147,16 +147,17 @@ export function useFetchTransactions(): QueryHookResult {
           const nonEmptyTransactions = returnedTransactions.filter(
             (returnedTransaction) => !isEmpty(returnedTransaction)
           )
+          const knownTransactionHashes = transactionHashesByNetwork[networkId]
           // Compare the new tx hashes with the ones we already have in redux
-          for (let i = 0; i < nonEmptyTransactions.length; i++) {
-            if (!transactionHashes.includes(nonEmptyTransactions[i].transactionHash)) {
+          for (const tx of nonEmptyTransactions) {
+            if (!knownTransactionHashes || knownTransactionHashes.has(tx.transactionHash)) {
```

Comment:
Should we check if tx hash in _not_ in known hashes?

```suggestion
            if (!knownTransactionHashes || !knownTransactionHashes.has(tx.transactionHash)) {
```

Reply:
lol yup, updated